pkg/search/search.go

```diff
@@ -0,0 +1,88 @@
+package search
+
+import (
+	"encoding/json"
+	//"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/maxott/magda-cli/pkg/adapter"
+	log "go.uber.org/zap"
+)
+
+/**** DATASETS ****/
+
+type DatasetRequest struct {
```

Comment:
You may want to change that. You actually don't request the dataset itself, only metadata about it. Maybe catalog search or something like that may be more descriptive.

Reply:
I named this because of Magda's API structure. For example, for records you have list, create, read, update and delete.

For search, you have datasets, facets, regions, organizations, etc. The URL for those is /search/datasets, /search/facets, /search/organizations, and so on... So this kind of naming will make it easier in the future to add these missing search endpoints to the CLI. I do see how it sounds a bit counterintuitive though. Thoughts?

Reply:
OK, sounds reasonable